File: labs/lab3/C++/main.cpp
#include <iostream>

using namespace std;

double chislitel(int a, int k, double x) {
    double result = 1;
    int i = 0;
    while (i <= (k + 1)) {
        i++;
        result = result * (a - i);
    }
    return result * pow(x, k);
}

double factorial(int k) {
    int n = 1;
    int i = 1;
    while (i <= k) {
        i++;
        n = n * i;
    }
    return n;
}

int main() {
    int a;
    cout << "Enter an integer number a = ";
    cin >> a;
    double x;
    cout << "Enter fractional x from 0 to 1, x = ";
    cin >> x;
    int k = 1;
    double sum = 1;
    bool finish = false;
    while (!finish) {
        sum = sum + chislitel(a, k, x) / factorial(k);
        k++;
        if (chislitel(a, k, x) == 0) finish = true;
    }
    printf("%.5f", sum);
    return 0;
}

File: labs/lab3/Python/main.py
def factorial(k):
    n = 1
    i = 1
    while i <= k:
        i = i + 1
        n = n * i
    return n


def chislitel(a, k, x):
    result = 1
    i = 0
    while i <= (k + 1):
        i = i + 1
        result = result * (a - i)
    return result * pow(x, k)


a = int(input('Enter an integer number, a = '))
x = float(input('Enter fractional x from 0 to 1, x = '))
k = 1
Sum: float = 1
finish = False
while (not finish):
    Sum = Sum + chislitel(a, k, x) / factorial(k)
    k = k + 1
    if chislitel(a, k, x) == 0:
        finish = True

print(round(Sum, 5))
